### ISSUE TITLE
iOS app crash when about:blank comes in app

### ISSUE THREAD

**Issue by alyabbasi93** [pichillilorenzo/flutter_inappwebview#1164]
iOS app crash when about:blank comes in app is any solution for that

**Comment by alyabbasi93** [comment]
I have an app when about:blank comes app crash is any way to ignore it or fix it

**Comment by pichillilorenzo** [comment]
Please post here crash logs and steps to reproduce the crash, thanks.

**Comment by alyabbasi93** [comment]
flutter: finalurlhttps://mtc-demo.space/explore/
flutter: finalurlhttps://mtc-demo.space/explore/
flutter: {message: JQMIGRATE: Migrate is installed, version 3.3.2, messageLevel: 1}
flutter: {message: google.maps.event.addDomListener() is deprecated, use the standard addEventListener() method instead: https://developer.mozilla.org/docs/Web/API/EventTarget/addEventListener
The feature will continue to work and there is no plan to decommission it., messageLevel: 2}
flutter: about:blank

**Comment by alyabbasi93** [comment]
flutter: about:blank make web view crash

**Comment by pichillilorenzo** [comment]
Just tested using the latest version of the plugin and URL as `about:blank` and it works correctly without any crash.

Here is the code example used:
```dart
import 'dart:async';
import 'dart:developer';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:flutter_inappwebview/flutter_inappwebview.dart';

Future main() async {
  WidgetsFlutterBinding.ensureInitialized();

  if (defaultTargetPlatform == TargetPlatform.android) {
    await AndroidInAppWebViewController.setWebContentsDebuggingEnabled(true);
  }

  runApp(const MaterialApp(
      home: MyApp()
  ));
}

class MyApp extends StatefulWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {

  final GlobalKey webViewKey = GlobalKey();

  InAppWebViewController? webViewController;
  InAppWebViewGroupOptions options = InAppWebViewGroupOptions(
      android: AndroidInAppWebViewOptions(
        useHybridComposition: true,
      ));

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(title: const Text("InAppWebView TEST about:blank")),
        body: SafeArea(
            child: Column(children: <Widget>[
              Expanded(
                child: Stack(
                  children: [
                    InAppWebView(
                      key: webViewKey,
                      initialUrlRequest: URLRequest(
                          url: Uri.parse('about:blank')),
                      initialOptions: options,
                      onWebViewCreated: (controller) {
                        webViewController = controller;
                      },
                      onLoadStart: (controller, url) {
                        log("onLoadStart $url");
                      },
                      onLoadStop: (controller, url) async {
                        log("onLoadStop $url");
                      },
                    ),
                  ],
                ),
              ),
            ]))
    );
  }
}
```

From what you posted, there is no error and no info about the crash.
If you are experiencing a crash, please post here the full log of the error!
Use XCode to run your app and take from there the stacktrace.

**Comment by alyabbasi93** [comment]
await widget.browser.open(
                              url: uri,
                              options: ChromeSafariBrowserClassOptions(
                                  android: AndroidChromeCustomTabsOptions(
                                      addDefaultShareMenuItem: false,
                                      keepAliveEnabled: true),
                                  ios: IOSSafariOptions(
                                      dismissButtonStyle:
                                          IOSSafariDismissButtonStyle.CLOSE,
                                      presentationStyle:
                                          IOSUIModalPresentationStyle
                                              .OVER_FULL_SCREEN)));
that why app crash

**Comment by alyabbasi93** [comment]
how to fix it

**Comment by alyabbasi93** [comment]
import 'dart:io';

import 'package:admob_flutter/admob_flutter.dart';
import 'package:fence/element/localnitification.dart';
import 'package:firebase_messaging/firebase_messaging.dart';
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:flutter_inappwebview/flutter_inappwebview.dart';
import 'package:permission_handler/permission_handler.dart';
import 'package:shared_preferences/shared_preferences.dart';
import 'package:qrscan/qrscan.dart' as scanner;

import '../../element/Loader.dart';

class MyChromeSafariBrowser extends ChromeSafariBrowser {
  @override
  void onOpened() {
    print("ChromeSafari browser opened");
  }

  @override
  void onCompletedInitialLoad() {
    print("ChromeSafari browser initial load completed");
  }

  @override
  void onClosed() {
    print("ChromeSafari browser closed");
  }
}

class webview2 extends StatefulWidget {
  final ChromeSafariBrowser browser = new MyChromeSafariBrowser();

  @override
  State<webview2> createState() => _webview2State();
}

class _webview2State extends State<webview2>
    with AutomaticKeepAliveClientMixin<webview2> {
  String view = '';
  var white = Colors.white;
  final GlobalKey webViewKey = GlobalKey();
  late AdmobInterstitial interstitialAd;
  GlobalKey<ScaffoldState> scaffoldState = GlobalKey();
  bool isLoading = true;
  String isViewed = '';

  InAppWebViewController? webViewController;
  InAppWebViewGroupOptions options = InAppWebViewGroupOptions(
      crossPlatform: InAppWebViewOptions(
          useShouldOverrideUrlLoading: true,
          mediaPlaybackRequiresUserGesture: false,
          cacheEnabled: true,
          javaScriptEnabled: true),
      android: AndroidInAppWebViewOptions(
        useHybridComposition: true,
      ),
      ios: IOSInAppWebViewOptions(
        allowsInlineMediaPlayback: true,
      ));

  late PullToRefreshController pullToRefreshController;
  String url = "";
  double progress = 0;
  final urlController = TextEditingController();

  void handleEvent(
      AdmobAdEvent event, Map<String, dynamic>? args, String adType) {
    switch (event) {
      case AdmobAdEvent.loaded:
        print('New Admob $adType Ad loaded!');
        break;
      case AdmobAdEvent.opened:
        print('Admob $adType Ad opened!');
        break;
      case AdmobAdEvent.closed:
        print('Admob $adType Ad closed!');
        break;
      case AdmobAdEvent.failedToLoad:
        print('Admob $adType failed to load. :(');
        break;
      case AdmobAdEvent.rewarded:
        showDialog(
          context: scaffoldState.currentContext!,
          builder: (BuildContext context) {
            return WillPopScope(
              onWillPop: () async {
                ScaffoldMessenger.of(context).hideCurrentSnackBar();
                return true;
              },
              child: AlertDialog(
                content: Column(
                  mainAxisSize: MainAxisSize.min,
                  children: <Widget>[
                    Text('Reward callback fired. Thanks Andrew!'),
                    Text('Type: ${args!['type']}'),
                    Text('Amount: ${args['amount']}'),
                  ],
                ),
              ),
            );
          },
        );
        break;
      default:
    }
  }

  getInterstitialAdUnitId() {
    if (Platform.isIOS) {
      return 'ca-app-pub-3940256099942544/4411468910';
    } else if (Platform.isAndroid) {
      return 'ca-app-pub-3940256099942544/1033173712';
    }
  }

  @override
  void initState() {
    // 1. This method call when app in terminated state and you get a notification
    // when you click on notification app open from terminated state and you can get notification data in this method

    super.initState();
    interstitialAd = AdmobInterstitial(
      adUnitId: getInterstitialAdUnitId(),
      listener: (AdmobAdEvent event, Map<String, dynamic>? args) {
        if (event == AdmobAdEvent.closed) interstitialAd.load();
        handleEvent(event, args, 'Interstitial');
      },
    );
    widget.browser.addMenuItem(ChromeSafariBrowserMenuItem(
        id: 1,
        label: 'Custom item menu 1',
        action: (url, title) {
          print('Custom item menu 1 clicked!');
          print(url);
          print(title);
        }));
    widget.browser.addMenuItem(ChromeSafariBrowserMenuItem(
        id: 2,
        label: 'Custom item menu 2',
        action: (url, title) {
          print('Custom item menu 2 clicked!');
          print(url);
          print(title);
        }));

    pullToRefreshController = PullToRefreshController(
      options: PullToRefreshOptions(
        color: Colors.blue,
      ),
      onRefresh: () async {
        if (Platform.isAndroid) {
          webViewController?.reload();
        } else if (Platform.isIOS) {
          webViewController?.loadUrl(
              urlRequest: URLRequest(url: await webViewController?.getUrl()));
        }
      },
    );
    interstitialAd.load();
  }

  initintads() async {
    final isLoaded = await interstitialAd.isLoaded;
    if (isLoaded ?? false) {
      interstitialAd.show();
    } else {
      print('Interstitial ad is still loading...');
    }
  }

  @override
  void dispose() {
    super.dispose();
  }

  Future _scan() async {
    await Permission.camera.request();
    String? barcode = await scanner.scan();
    if (barcode == null) {
      print('nothing return.');
    } else {
      webViewController?.loadUrl(
          urlRequest: URLRequest(url: Uri.parse(barcode)));
    }
  }

  @override
  Widget build(BuildContext context) {
    //initintads();
    return WillPopScope(
      onWillPop: () async => false,
      child: Scaffold(
          floatingActionButton: Container(
            padding: EdgeInsets.only(top: 50.0),
            child: Align(
              alignment: Alignment.topLeft,
              child: FloatingActionButton(
                backgroundColor: Colors.transparent,
                onPressed: () {
                  setState(() {
                    webViewController!.goBack();
                  });
                },
                child: Icon(Icons.arrow_back),
              ),
            ),
          ),
          body: SafeArea(
              child: Column(children: <Widget>[
            Expanded(
              child: Stack(
                children: [
                  WillPopScope(
                    onWillPop: () async => false,
                    child: InAppWebView(
                      key: webViewKey,
                      initialUrlRequest: URLRequest(
                          url: Uri.parse('https://app.fence.lu/main')),
                      initialOptions: options,
                      pullToRefreshController: pullToRefreshController,
                      onWebViewCreated: (controller) {
                        webViewController = controller;
                      },
                      onLoadStart: (controller, url) {
                        setState(() {
                          this.url = url.toString();
                          urlController.text = this.url;
                        });
                      },
                      androidOnPermissionRequest:
                          (controller, origin, resources) async {
                        return PermissionRequestResponse(
                            resources: resources,
                            action: PermissionRequestResponseAction.GRANT);
                      },
                      shouldOverrideUrlLoading:
                          (controller, navigationAction) async {
                        var uri = navigationAction.request.url!;
                        print(uri.path);

                        if (uri.path.contains('/openscanner')) {
                          setState(() {
                            _scan();
                          });
                        }
                        if (uri.path.contains('/openscanner')) {}
                        if (!uri.host.contains('app.fence.lu')) {
                          await widget.browser.open(
                              url: uri,
                              options: ChromeSafariBrowserClassOptions(
                                  android: AndroidChromeCustomTabsOptions(
                                      addDefaultShareMenuItem: false,
                                      keepAliveEnabled: true),
                                  ios: IOSSafariOptions(
                                      dismissButtonStyle:
                                          IOSSafariDismissButtonStyle.CLOSE,
                                      presentationStyle:
                                          IOSUIModalPresentationStyle
                                              .OVER_FULL_SCREEN)));

                          return NavigationActionPolicy.CANCEL;
                        }

                        return NavigationActionPolicy.ALLOW;
                      },
                      onLoadStop: (controller, url) async {
                        pullToRefreshController.endRefreshing();
                        setState(() {
                          this.url = url.toString();
                          urlController.text = this.url;
                        });
                      },
                      onLoadError: (controller, url, code, message) {
                        pullToRefreshController.endRefreshing();
                      },
                      onProgressChanged: (controller, progress) {
                        if (progress == 100) {
                          pullToRefreshController.endRefreshing();
                        }
                        setState(() {
                          this.progress = progress / 100;
                          urlController.text = this.url;
                        });
                      },
                      onUpdateVisitedHistory:
                          (controller, url, androidIsReload) {
                        setState(() {
                          this.url = url.toString();
                          urlController.text = this.url;
                        });
                      },
                      onConsoleMessage: (controller, consoleMessage) {
                        print(consoleMessage);
                      },
                    ),
                  ),
                  progress < 1.0
                      ? LinearProgressIndicator(value: progress)
                      : Container(),
                ],
              ),
            ),
          ]))),
    );
  }

  Widget _buildProgressBar() {
    if (progress != 1.0) {
      return Loader(
        type: 'DoubleBounce',
        color: Colors.black26,
      );
// You can use LinearProgressIndicator also
//      return LinearProgressIndicator(
//        value: progress,
//        valueColor: new AlwaysStoppedAnimation<Color>(Colors.orange),
//        backgroundColor: Colors.blue,
//      );
    }
    return Container();
  }

  saveurl(String url) async {
    SharedPreferences pref = await SharedPreferences.getInstance();
    await pref.setString('url', url);
  }

  loadurl() async {
    SharedPreferences pref = await SharedPreferences.getInstance();
    view = pref.getString('url') ?? '';
    return view;
  }

  @override
  bool get wantKeepAlive => true;
}

**Comment by pichillilorenzo** [comment]
Ok, I found that [iOS SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller/1649141-init) supports only `http` or `https` scheme.
The app crashes with the following error from XCode:
```
Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'The specified URL has an unsupported scheme. Only HTTP and HTTPS URLs are supported.'
```

So, `about:blank` is not a valid URL for `ChromeSafariBrowser` on iOS.
I will update the docs and add an `assert()` inside the `open()` method to throw an error on iOS so you can't use URL that doesn't have `http` or `https` as scheme.

In this case you should check if the `uri.scheme` is `http` or `https`, at least on iOS platform.

**Comment by pichillilorenzo** [comment]
Released new version `5.4.3+4`.
Now it throws an error if you try to open `ChromeSafariBrowser` on iOS using a different scheme than `http` and `https`
